content/en/news/releases/1.9.x/announcing-1.9/upgrade-notes/index.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Upgrade Notes
+description: Important changes to consider when upgrading to Istio 1.9.0.
+weight: 20
+---
+
+When you upgrade from Istio 1.8 to Istio 1.9.0, you need to consider the changes on this page.
+These notes detail the changes which purposefully break backwards compatibility with Istio 1.8.
+The notes also mention changes which preserve backwards compatibility while introducing new behavior.
+Changes are only included if the new behavior would be unexpected to a user of Istio 1.8.
+
+## PeerAuthentication per-port-level configuration will now also apply to pass through filter chains.
+Previously the PeerAuthentication per-port-level configuration is ignored if the port number is not defined in a
+service and the traffic will be handled by a pass through filter chain. Now the per-port-level setting will be
+supported even if the port number is not defined in a service, a special pass through filter chain will be added
+to respect the corresponidng per-port-level mTLS specification.
+Pleae check your PeerAuthentication to make sure you are not using the per-port-level configuration on pass through
```

Comment:
Do we have any tool (istioctl analyzer) users can run to detect this?

Reply:
We do not. I don't think `analyze` is the right place for this if we do add it. This is strictly a bug fix - but one that users *might* have depended on.

Reply:
Ok, I agree this is a rare case.  I was referring to a tool to help users to detect bad configs, e.g. peer authentication has per-port-level configuration but the port was not defined in the service.

Reply:
This item is tickling a "is it backwards compatible" question in my head. We're changing the behavior of a (previously undocumented) combination of configuration. What is the chance that a customer will experience breakage because of this change? @smawson PTAL.

Reply:
> I was referring to a tool to help users to detect bad configs,

The whole point of the change is that this is *not* a bad config. Its perfectly valid to apply a rule to port 12345 and not have it in the service - this is exactly why we made the beta security APIs. Security policies apply to workloads, not services.

> What is the chance that a customer will experience breakage because of this change?

This would only happen if a user explicitly wrote a configuration that did absolutely nothing, and kept it around for whatever reason. I am sure this is a non-zero set of people, whether from applying random things or other cases, but there is no legitimate use case for wanting the old behavior, as the old behavior was "this configuration doesn't actually apply" - if a user doesn't want it to apply they should just not have the config at all.

In theory all bug fixes are breaking changes :slightly_smiling_face:

Reply:
To clarify: basically any bug we fix is a breaking change to some degree. Often these are so obviously "worth it" like "make istiod not crash" - no user reasonably *wants* istiod to crash. Others are potentially quite disruptive, like if we fixed some bug by changing core routing logic that impacts many services, for example.

To me, this one feels pretty far on the side of not too painful - in the rare case that a user was impacted by it, I think they would agree the change here makes sense.

My favorite comic: https://xkcd.com/1172/

Reply:
Thanks, @howardjohn for the details.

Reply:
Marking it resolve as it was a question and I assume no change required in release notes.